### ISSUE TITLE
No response for await Voximplant.getInstance().call() sometimes

### ISSUE THREAD

**Issue by justsomeofl** [voximplant/react-native-voximplant#129]
Sometimes the app gets stuck on this line (so impossible to make calls):
let call = await Voximplant.getInstance().call(this.callTo, callSettings);

This line is called when i press the make call button. Sometimes i press the button and there is nothing happening.
Most of the times the line works but 1/5 times there is no response and once there is no response it stay like this until the app is killed and started again. 

It happens on both Android and iOS.

I use the latest versions of react-native-voximplant and SDK.

Any suggestion ?
Thanks

**Comment by YuliaGrigorieva** [comment]
Hello @justsomeofl ,

Could you please collect the adb logcat on android (with `VOXSDK` tag) or Xcode output logs (with `#VI` tag)? It will help to analyze the issue and find the root cause.

Best regards,
Yulia Grigorieva

**Comment by justsomeofl** [comment]
Hello @YuliaGrigorieva, 

I tried multiple calls and it is looping on the same logs:
```
2020-11-11 06:11:45.253+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:11:45.348+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
2020-11-11 06:11:55.458+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:11:56.001+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
2020-11-11 06:12:06.021+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:12:06.052+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
2020-11-11 06:12:16.058+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:12:16.118+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
2020-11-11 06:12:26.223+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:12:26.276+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
2020-11-11 06:12:36.315+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > ===> {"name":"__ping"} 
2020-11-11 06:12:36.392+0100 Konekt #VI/I [signaling ]:VISignaling[Connected] > <=== {"name" : "__pong" , "params" : []} 
```
And after 15 minutes i got different logs:
```
2020-11-11 06:27:08.795+0100 Konekt #VI/E [signaling ]:VISignaling[Connected] > webSocket:didFailWithError: Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={_kCFStreamErrorCodeKey=57, _kCFStreamErrorDomainKey=1} 
2020-11-11 06:27:08.798+0100 Konekt #VI/I [signaling ]:VISignaling[Disconnected] > closeConnection 
2020-11-11 06:27:08.798+0100 Konekt #VI/E [sm        ]:VIAuthenticator[LoggedIn] > webSocketDidCloseWithError: Error Domain=com.voximplant.connectivity Code=10001 "The operation couldn’t be completed. Socket is not connected" UserInfo={NSUnderlyingError=0x283f864f0 {Error Domain=NSPOSIXErrorDomain Code=57 "Socket is not connected" UserInfo={_kCFStreamErrorCodeKey=57, _kCFStreamErrorDomainKey=1}}, NSLocalizedDescription=The operation couldn’t be completed. Socket is not connected, reason=The operation couldn’t be completed. Socket is not connected} 
2020-11-11 06:27:08.799+0100 Konekt #VI/I [sm        ]:VICallManager > endAllCalls: (
2020-11-11 06:27:08.799+0100 Konekt #VI/I [sm        ]:VICall[03495D18645477EC.1605071426.1254659:Connected] > forceStop 
2020-11-11 06:27:08.799+0100 Konekt #VI/I [sm        ]:VIAuthenticator[Disconnected] > webSocketConnectionDidClose 
2020-11-11 06:27:08.799+0100 Konekt #VI/I [sm        ]:VICallManager > endAllCalls: (
2020-11-11 06:27:08.799+0100 Konekt #VI/I [sm        ]:VICall[03495D18645477EC.1605071426.1254659:Connected] > forceStop 
2020-11-11 06:27:08.816+0100 Konekt #VI/I [overcommit]:VIAudioManager > audioSessionDidChangeRoute: [1] 1 -> 0 CategoryChange listevent:V devevent:X0 (categories: AVAudioSessionCategoryPlayAndRecord -> AVAudioSessionCategoryPlayAndRecord) 
2020-11-11 06:27:08.820+0100 Konekt #VI/W [overcommit]:WebRTC > (pacing_controller.cc:239): Elapsed time (330811 ms) longer than expected, limiting to 2000 
2020-11-11 06:27:09.336+0100 Konekt #VI/I [main-threa]:<VIAudioSessionOverlapNotifier: 0x283220670> > audioSessionDidEndInterruption:shouldResumeSession: YES 
2020-11-11 06:27:10.108+0100 Konekt #VI/I [overcommit]:VIAudioManager > audiosession did change route begin... 
2020-11-11 06:27:10.170+0100 Konekt #VI/I [overcommit]:VIAudioManager > audioSessionDidChangeRoute: [1] 1 -> 1 CategoryChange listevent:V devevent:V0 (categories: AVAudioSessionCategoryPlayAndRecord -> AVAudioSessionCategoryPlayAndRecord) 
2020-11-11 06:27:10.187+0100 Konekt #VI/I [overcommit]:VIAudioManager > audiosession did change route begin... 
2020-11-11 06:27:10.202+0100 Konekt #VI/I [overcommit]:VIAudioManager > audioSessionDidChangeRoute: [1] 0 -> 1 CategoryChange listevent:V devevent:V0 (categories: AVAudioSessionCategoryPlayAndRecord -> AVAudioSessionCategoryPlayAndRecord) 
2020-11-11 06:27:10.205+0100 Konekt #VI/I [overcommit]:VIAudioManager > audiosession did change route begin... 
2020-11-11 06:27:10.215+0100 Konekt #VI/I [overcommit]:VIAudioManager > audioSessionDidChangeRoute: [1] 1 -> 1 CategoryChange listevent:V devevent:V0 (categories: AVAudioSessionCategoryPlayAndRecord -> AVAudioSessionCategoryPlayAndRecord) 
```

I also tried to add login to Voximplant code (in case the VI user was logged out) before each call but same thing.
Maybe there is  a way to reset Voximplant as if the app is killed (cause after the app is killed it works again) ?

Thanks

**Comment by YuliaGrigorieva** [comment]
Hello @justsomeofl ,

According to these logs snippets, there were no calls (1st snippet) but the SDK was connected to the Voximplant Cloud, then (2nd snippet) the connection to the Voximplant Cloud was closed probably due to a network issue and there was an active call at this moment that is also disconnected (it is expected).

Unfortunately, these logs do not contain the information about how the call was created, so I can't say anything about the issue.
I do not currently see any signs that Voximplant SDK behaves incorrectly and requires a reset.

I recommend to collect full logs from the app launch til the issue with the Client.call API (even they are long) and send them to our support team at support@voximplant.com

Best regards,
Yulia Grigorieva

**Comment by justsomeofl** [comment]
Thanks, i've sent an email to support@voximplant.com with full logs

**Comment by justsomeofl** [comment]
Issue fixed since the update to 1.24.0
